portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Policies/Policies.tsx

```diff
@@ -175,53 +177,83 @@ const Policies: React.FC = () => {
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name + obj.version, obj), new Map()).values()];
+                commonPolicyByPolicyDisplayName = [...commonPoliciesList
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name + obj.version, obj), new Map()).values()];
             } else {
-                // Get the union of policies depending on the policy display name
-                unionByPolicyDisplayName = [...mergedList
-                    .reduce((map, obj) => map.set(obj.name, obj), new Map()).values()];
+                // Get the policies depending on the policy display name
+                apiPolicyByPolicyDisplayName = [...apiSpecificPoliciesList
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name, obj), new Map()).values()];
+                commonPolicyByPolicyDisplayName = [...commonPoliciesList
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name, obj), new Map()).values()];
             }
-            unionByPolicyDisplayName.sort(
+            apiPolicyByPolicyDisplayName.sort(
                 (a: Policy, b: Policy) => a.name.localeCompare(b.name))
+            commonPolicyByPolicyDisplayName.sort(
+                (a: Policy, b: Policy) => a.name.localeCompare(b.name))
+
+            let filteredApiPolicyByGatewayTypeList = null;
+            let filteredCommonPolicyByGatewayTypeList = null;
             
-            let filteredByGatewayTypeList = null;
             if (!isChoreoConnectEnabled) {
                 // Get synpase gateway supported policies
-                filteredByGatewayTypeList = unionByPolicyDisplayName.filter(
+                filteredApiPolicyByGatewayTypeList = apiPolicyByPolicyDisplayName.filter(
+                    (policy: Policy) => policy.supportedGateways.includes('Synapse'));
+                filteredCommonPolicyByGatewayTypeList = commonPolicyByPolicyDisplayName.filter(
                     (policy: Policy) => policy.supportedGateways.includes('Synapse'));
             } else {
                 // Get CC gateway supported policies
-                filteredByGatewayTypeList = unionByPolicyDisplayName.filter(
+                filteredApiPolicyByGatewayTypeList = apiPolicyByPolicyDisplayName.filter(
+                    (policy: Policy) => policy.supportedGateways.includes('ChoreoConnect'));
+                filteredCommonPolicyByGatewayTypeList = commonPolicyByPolicyDisplayName.filter(
                     (policy: Policy) => policy.supportedGateways.includes('ChoreoConnect'));
             }
 
-            let filteredByAPITypeList = null;
+            let filteredApiPoliciesByAPITypeList = [];
+            let filteredCommonPoliciesByAPITypeList = [];
+            
             if (api.type === "HTTP") {
```

Comment:
This can be simplify as,
```js
if (api.type === 'HTTP' || api.type === 'SOAP' || api.type === 'SOAPTOREST') {
    // Get HTTP, SOAP and SOAP to REST supported policies
    filteredApiPoliciesByAPITypeList = filteredApiPolicyByGatewayTypeList.filter(
        (policy: Policy) => policy.supportedApiTypes.includes(api.type));
    filteredCommonPoliciesByAPITypeList = filteredCommonPolicyByGatewayTypeList.filter(
        (policy: Policy) => policy.supportedApiTypes.includes(api.type));
}
```